kedro/contrib/io/yaml_local/yaml_local.py

```diff
@@ -0,0 +1,119 @@
+# BE LIABLE FOR ANY CLAIM, DAMAGES, OR OTHER LIABILITY, WHETHER IN AN
+# ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF, OR IN
+# CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+#
+# The QuantumBlack Visual Analytics Limited ("QuantumBlack") name and logo
+# (either separately or in combination, "QuantumBlack Trademarks") are
+# trademarks of QuantumBlack. The License does not grant you any right or
+# license to the QuantumBlack Trademarks. You may not use the QuantumBlack
+# Trademarks or any confusingly similar mark as a trademark for your product,
+#     or use the QuantumBlack Trademarks in any other manner that might cause
+# confusion in the marketplace, including but not limited to in advertising,
+# on websites, or on software.
+#
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""``YAMLLocalDataset`` loads and saves data to a local yaml file using
+``PyYAML``.
+See https://pyyaml.org/wiki/PyYAMLDocumentation for details.
+"""
+import copy
+from pathlib import Path
+from typing import Any, Dict
+
+import pandas as pd
+import yaml
+
+from kedro.io.core import AbstractVersionedDataSet, DataSetError, Version
+
+
+class YAMLLocalDataSet(AbstractVersionedDataSet):
+    """
+    ``YAMLLocalDataset`` loads and saves data to a local yaml file using
+    ``PyYAML``.
+    See https://pyyaml.org/wiki/PyYAMLDocumentation for details.
+
+    Example:
+    ::
+
+        >>> from kedro.contrib.io.yaml_local import YAMLLocalDataSet
+        >>> my_dict = {
+        >>>     'a_string': 'Hello, World!',
+        >>>     'a_list': [1, 2, 3]
+        >>> }
+        >>> data_set = YAMLLocalDataSet(filepath="test.yml")
+        >>> data_set.save(my_dict)
+        >>> reloaded = data_set.load()
+        >>> assert my_dict == reloaded
+
+    """
+
+    DEFAULT_SAVE_ARGS = {"default_flow_style": False}  # type: Dict[str, Any]
+
+    def __init__(
+        self,
+        filepath: str,
+        save_args: Dict[str, Any] = None,
+        version: Version = None,
+    ) -> None:
+        """Creates a new instance of ``YAMLLocalDataset`` pointing to a concrete
+        filepath.
+
+        Args:
+            filepath: path to a local yaml file.
+            save_args: Arguments passed on to ```yaml.dump``.
+                See https://pyyaml.org/wiki/PyYAMLDocumentation for details.
+                ``{"default_flow_style": False}`` in default.
+            version: If specified, should be an instance of
+                ``kedro.io.core.Version``. If its ``load`` attribute is
+                None, the latest version will be loaded. If its ``save``
+                attribute is None, save version will be autogenerated.
+
+        """
+        super().__init__(Path(filepath), version)
+
+        # Handle default load and save arguments
+        self._save_args = copy.deepcopy(self.DEFAULT_SAVE_ARGS)
```

Comment:
Please look at the previous comments https://github.com/quantumblacklabs/kedro/pull/66/files/98aad7acf04193d85d60012a788302772fc540a7#r311487946
We haven't done this for the core data sources, but still please keep the contrib package consistent.

Reply:
OK.